### PR TITLE
Feat/reportmodal

### PR DESCRIPTION
**Pull request by kentacohut**
@danieljhegeman Could you take a look at this for me? I finally got my bugs worked out with the report photo modal. 

screen shot 2018-03-03 at 3 18 40 pm

### CODE REVIEW

File: client/src/photos.jsx [hunk 1]
```diff
@@ -10,43 +11,33 @@ class App extends React.Component {
...
-
-  closeReportPhoto() {
-    this.setState({
-      reportPhotoIsOpen: false
-    })
-  }
-
   gotoPrevious() {
     this.setState({
       currentImage: this.state.currentImage - 1,
```
Comment (danieljhegeman) [inline client/src/photos.jsx]: What happens when this hits zero? ( same for gotoNext() )

File: client/src/photos.jsx [hunk 2]
```diff
@@ -63,55 +54,116 @@ class App extends React.Component {
...
+  componentDidMount() {
+    let id = window.location.href.split('/').pop().substring(4) || '1'; //Grabs params from url 
     this.findPhotos(id);
   }
 
-  componentDidUpdate(){
-    $('div[class^="footerCount"]').on('click', ()=>{this.openReportPhoto()});
+  componentDidUpdate() {
+    setTimeout(() => { //Allows jQuery to grab report flag from lightbox footer
+      $('div[class^="footerCount"]').on('click', ()=>{ this.handleOpenModal(); });
```
Comment (danieljhegeman) [inline client/src/photos.jsx]: Why are you using a setTimeout when you are already firing this post-update? Are you trying to make sure it attaches event-handlers *after* the images from an http request have loaded? If this is circumventing some sort of "element not on the DOM yet" problem, then it seems practical to instead employ a function that is triggered upon complete loading of all images and use that function to immediately attach event-handlers.

Though I could be reading the purpose of your setTimeout maneuver incorrectly.

Lastly, good job using the arrow function to preserve 'this' context.

File: client/src/photos.jsx (hunk 1 above)
Comment (kentacohut) [reply client/src/photos.jsx]: When it get's to zero or the max length of the photos array the previous or next arrows will not appear.

File: client/src/photos.jsx (hunk 2 above)
Comment (kentacohut) [reply client/src/photos.jsx]: Yes, what happens is that the event listener is triggered with state change but the DOM has not finished fully rendering yet. I was trying to develop a function like you suggest but had no luck. Maybe we can discuss this and you can give me some pointers. Thanks!

**Comment by kentacohut** [comment]
@danieljhegeman 
Hey Dan, I made the suggested changes. I hope this appeases you.
Sincerely,
Steve

**Comment by danieljhegeman** [comment]
Hello @kentacohut!

Great job attaching the event listener to the flag on the photo modal without resorting to setTimeout. Merge approved.

**Pull request merged**